cmd/manager/main.go

```diff
@@ -146,10 +150,17 @@ func start() error {
 		WithServiceName(operatorconfig.OperatorName).
 		GetConfig()
 
-	// Configure metrics if it errors log the error but continue
-	if err := metrics.ConfigureMetrics(context.TODO(), *metricsServer); err != nil {
-		log.Error(err, "Failed to configure Metrics")
-		os.Exit(1)
+	// detect if operator-sdk up local is run
+	detectSDKLocal := os.Getenv(envSDK)
+
+	// Configure metrics. If it errors, log the error and exit.
+	if detectSDKLocal == "local" {
+		log.Info("Skipping metrics configuration; not running in a cluster.")
+	} else {
+		if err := metrics.ConfigureMetrics(context.TODO(), *metricsServer); err != nil {
+			log.Error(err, "Failed to configure Metrics")
+			os.Exit(1)
```

Comment:
return err

Reply:
and replace top ones too to `returun err`

Reply:
@mjudeikis Do you mean replace all the `os.Exit(1)` to `return err` in this `start()` func?

Reply:
yes